feat: add dark mode

alphatrl opened the pull request:
Add dark mode for react-native-emoji-selector

- Add dark mode
    - search bar and text switches colour
    - update example to support dark mode 
- Change default colours to use iOS platform colours (switches colour when alternating between dark and light mode)
- Add devdependencies modules to improve code readability
- Update example/package.json to use local module instead of copying and paste
- Add proptypes

alphatrl committed:
chore: update formatting

alphatrl committed:
feat: support dark mode within expo app

- update expo version
- use react-native appearance hooks
- rewrite app to functional component
- add dark mode colour

alphatrl committed:
feat: dark mode support for emoji component

- dark mode support for search bar (iOS) and search bar text (iOS and android)
- split searchbar into its own functional component
- remove search bar bgColor
- fix android appearance

alphatrl committed:
chore: update package.json dependencies to peerdependencies

alphatrl committed:
feat: add proptypes validation and use local module

- add proptype validation
- update example package.json to use local module

lxcid commented on LICENSE.md:
```diff
@@ -1,7 +1,7 @@
 =====================
 
-Copyright © `2019` `Arron Hunt <arronjhunt@gmail.com>`
+Copyright © `2021` `Taskade Inc <amos@taskade.com>`
```
Let's keep the original author credit and don't overwrite it. We can addon if we want, but not replace the original author.

lxcid commented on package.json:
```diff
@@ -11,13 +11,30 @@
     "react"
   ],
-  "author": "Arron Hunt",
+  "author": "Amos Tan",
```
Same thing here, shouldn't replace original author, we can add contributors if we want.

See https://docs.npmjs.com/cli/v6/configuring-npm/package-json#people-fields-author-contributors

lxcid commented on package.json:
```diff
@@ -11,13 +11,30 @@
   "license": "MIT",
-  "dependencies": {
-    "emoji-datasource": "^6.0.0"
+  "peerDependencies": {
```
go with normal `dependencies`, `peerDependencies` is a bit weird for consuming, normal dependencies should be fine for this case.

alphatrl committed:
fix: update configs

elliscwc commented on example/App.js:
```diff
@@ -1,41 +1,51 @@
-  }
+export default function App() {
+  const [emoji, setEmoji] = useState(" ");
+  const isDarkMode = useColorScheme() === 'dark';
```
we should use app provided dark mode setting in case user choose to set it manually

alphatrl replied:
sure, make sense!

alphatrl replied:
Should dark-mode be set manually, `DynamicColorIOS` should not be used due to possible UI issues. For example, a manually set dark-mode with a light system theme, will display a lightly coloured search bar against a white background resulting in low contrast.

Setting the styles manually will overcome these limitations

alphatrl committed:
feat: add dark mode prop to toggle between light and dark

alphatrl committed:
refactor: remove platformcolor

- remove platformcolor to support older version of react-native
- update readme.md instructions

elliscwc commented:
LGTM!
